src/main/java/tuteez/model/tag/Lesson.java

```diff
@@ -0,0 +1,92 @@
+package tuteez.model.tag;
+
+
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+
+import static java.util.Objects.requireNonNull;
+import static tuteez.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Lesson in the address book.
+ * Guarantees: immutable; day and time are valid as declared in {@link #isValidLesson(String, String)}
+ */
+public class Lesson {
```

Comment:
I believe you are missing a method which is an API call to check if a lesson already exists. This is to provide a warning to users if they choose to add a duplicate lesson